pilopt/src/lib.rs

```diff
@@ -427,6 +435,86 @@ fn simplify_expression_single<T: FieldElement>(e: &mut AlgebraicExpression<T>) {
     }
 }
 
+/// Uses associative properties to simplify expressions by regrouping constant terms.
+fn try_simplify_associative_operation<T: FieldElement>(
```

Comment:
The signature still looks a bit weird to me - how can you take two mutable references and return a new value?
I think it would be cleaner to just take a mutable reference to an `Expression` and return nothing (or a bool).

Otherwise you would have to explain what happens to `left` and `right` after the function call.

Reply:
Or, if that works with the returned value, take a mutable ref to a BinaryOperation.